tests/test_resources/test_modules/test_functions/test_function.py

```diff
@@ -173,301 +75,600 @@ async def async_slow_generator(size):
         yield f"Hello from the cluster! {arr}"
 
 
-@pytest.mark.clustertest
-@pytest.mark.asyncio
-async def test_async_generator(cluster):
-    remote_slow_generator = rh.function(async_slow_generator).to(cluster)
-    results = []
-    async for val in remote_slow_generator(5):
-        assert val
-        print(val)
-        results += [val]
-    assert len(results) == 5
-
-
-@pytest.mark.clustertest
-def test_remotes(cluster):
-    pid_fn = rh.function(getpid, system=cluster)
-
-    pid_key = pid_fn.run()
-    pid_res = cluster.get(pid_key).fetch()
-    assert pid_res > 0
-
-    # Test passing a remote into a normal call
-    pid_blob = pid_fn.remote()
-    pid_res = cluster.get(pid_blob.name).fetch()
-    assert pid_res > 0
-    pid_res = pid_blob.fetch()
-    assert pid_res > 0
-
-
-@pytest.mark.clustertest
-def test_function_git_fn(cluster):
-    remote_parse = rh.function(
-        fn="https://github.com/huggingface/diffusers/blob/"
-        "main/examples/dreambooth/train_dreambooth.py:parse_args",
-        system=cluster,
-        env=[
-            "torch==1.12.1 --verbose",
-            "torchvision==0.13.1",
-            "transformers",
-            "datasets",
-            "evaluate",
-            "accelerate",
-            "pip:./diffusers --verbose",
-        ],
-    )
-    args = remote_parse(
-        input_args=[
-            "--pretrained_model_name_or_path",
-            "stabilityai/stable-diffusion-2-base",
-            "--instance_data_dir",
-            "remote_image_dir",
-            "--instance_prompt",
-            "a photo of sks person",
-        ]
-    )
-    assert args.pretrained_model_name_or_path == "stabilityai/stable-diffusion-2-base"
-
-
-@pytest.mark.clustertest
-def test_list_keys(cluster):
-    pid_fn = rh.function(getpid).to(system=cluster)
-
-    pid_obj1 = pid_fn.run()
-    pid_obj2 = pid_fn.run()
-
-    current_jobs = cluster.keys()
-    assert set([pid_obj1, pid_obj2]).issubset(current_jobs)
-
-    pid_obj3 = pid_fn.remote()
-    pid_obj4 = pid_fn.remote()
-
-    current_jobs = cluster.keys()
-    assert set([pid_obj3.name, pid_obj4.name]).issubset(current_jobs)
-
-
 def slow_getpid(a=0):
     time.sleep(10)
     return os.getpid() + a
 
 
-@pytest.mark.clustertest
-def test_cancel_jobs(cluster):
-    pid_fn = rh.function(slow_getpid).to(cluster)
-
-    pid_run1 = pid_fn.run(2)
-
-    time.sleep(1)  # So the runkeys are more than 1 second apart
-    pid_ref2 = pid_fn.run(5)
+class TestFunction:
+
+    # ---------- Minimal / Local Level Tests (aka not unittests) ----------
+
+    @pytest.mark.rnstest
+    @pytest.mark.level("local")
+    @pytest.mark.usefixtures("cluster")
```

Comment:
You can take these off each function and instead apply it once to the class above:
```
@pytest.mark.usefixtures("cluster")
class TestFunction:
```